gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py

```diff
@@ -0,0 +1,424 @@
+
+""" Implements test cases for driver methods
+of generic config.
+
+"""
+
+
+class VpnGenericConfigDriverTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(VpnGenericConfigDriverTestCase, self).__init__(*args, **kwargs)
+
+        self.dict_objects = test_vpn_data.MakeDictionaries()
+        self.context = self.dict_objects._make_service_context()
+        self.plugin_rpc = vpn.VpnaasRpcSender(self.context,
+                                              self.dict_objects.sc)
+        self.rest_apt = vyos_vpn_driver.RestApi(self.dict_objects.vm_mgmt_ip)
+        self.driver = vyos_vpn_driver.VpnGenericConfigDriver()
+        self.resp = mock.Mock()
+        self.fake_resp_dict = {'status': True}
+        self.kwargs = self.dict_objects._fake_kwargs()
+
+    def setUp(self):
+        self.resp = mock.Mock(status_code=200)
+
+    def tearDown(self):
+        self.resp = mock.Mock(status_code=200)
+
+    def test_configure_interfaces(self):
+        """ Implements test case for configure interfaces method
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        with mock.patch.object(
+                requests, 'post', return_value=self.resp) as mock_post, \
+            mock.patch.object(self.resp,
+                              'json',
+                              return_value=self.fake_resp_dict):
+            self.driver.configure_interfaces(self.dict_objects.context_device,
+                                             self.kwargs)
+
+            mock_post.assert_called_with(
+                            self.dict_objects.url_for_add_inte,
+                            json.dumps(self.dict_objects.data_for_interface),
+                            timeout=self.dict_objects.timeout)
+
+    def test_clear_interfaces(self):
+        """ Implements test case for clear interfaces method
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        self.resp = mock.Mock(status_code=200)
+        with mock.patch.object(
+                requests, 'delete', return_value=self.resp) as mock_delete, \
+            mock.patch.object(
+                self.resp, 'json', return_value=self.fake_resp_dict):
+            self.driver.clear_interfaces(self.dict_objects.context_device,
+                                         self.kwargs)
+
+            mock_delete.assert_called_with(
+                        self.dict_objects.url_for_del_inte,
+                        data=json.dumps(self.dict_objects.data_for_interface),
+                        timeout=self.dict_objects.timeout)
+
+    def test_configure_source_routes(self):
+        """ Implements test case for configure routes method
+        of generic config driver.
+
+        Returns: none
+
+        """
+
+        with mock.patch.object(
+                requests, 'post', return_value=self.resp) as mock_post, \
+            mock.patch.object(json, 'loads',
```

Comment:
since we are mocking the post method we will not get the response so just seed the status of the request we need to mock as already one value seeded before this.